### PR TITLE
Support `stype.embedding`

### PR DESCRIPTION
**Pull request by akihironitta**
Part of #88.

### CODE REVIEW

**Comment by akihironitta** [comment]
This PR is on-hold until advanced indexing is supported in `MultiEmbeddingTensor`.

File: torch_frame/data/mapper.py [hunk 1]
```diff
@@ -395,3 +396,31 @@ def forward(
...
+    def forward(
+        self,
+        ser: Series,
+        *,
+        device: Optional[torch.device] = None,
+    ) -> MultiEmbeddingTensor:
+        # NOTE We are converting the default PyTorch dtype into a string
+        # representation that can be understood by numpy.
+        # TODO Think of a "less hacky" way to do this.
+        dtype = str(torch.get_default_dtype()).split('.')[-1]
```
Comment (weihua916) [inline torch_frame/data/mapper.py:412]: For embedding, the dtype should always be float32?

File: torch_frame/data/mapper.py [hunk 2]
```diff
@@ -395,3 +396,31 @@ def forward(
...
+        values = ser.explode().astype(dtype).values
+        values = torch.from_numpy(values).view(len(ser), -1)
```
Comment (weihua916) [inline torch_frame/data/mapper.py:414]:
```python
values = torch.from_numpy(np.stack(df.values))
```

File: torch_frame/data/stats.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ class StatType(Enum):
 
     # text_embedded (Also, embedding)
     # Note: For text_embedded, this stats is computed in
-    # dataset.update_col_stats, not here.
+    # dataset._update_col_stats, not here.
```
Comment (weihua916) [inline torch_frame/data/stats.py:42]: We can follow-up in the next PR, but we need to compute `EMB_DIM` as `compute_stats` here for `stype.embedding`.

**Review by weihua916 — APPROVED** [review]
Looks great, thanks!

Left a comment on follow-up PR for `EMB_DIM` stats computation.

File: torch_frame/data/mapper.py (hunk 1 above)
Comment (zechengz) [reply torch_frame/data/mapper.py:412]: Maybe we should keep this if user want to train the model with float16?

File: torch_frame/data/tensor_frame.py [hunk 4]
```diff
@@ -186,6 +187,12 @@ def __eq__(self, other: Any) -> bool:
                 if not MultiNestedTensor.allclose(self_feat, other_feat,
                                                   equal_nan=True):
                     return False
+            elif isinstance(self_feat, MultiEmbeddingTensor):
+                if not isinstance(other_feat, MultiEmbeddingTensor):
+                    return False
+                if not MultiNestedTensor.allclose(self_feat, other_feat,
```
Comment (zechengz) [inline torch_frame/data/tensor_frame.py:193]: Maybe this?
```suggestion
                if not MultiEmbeddingTensor.allclose(self_feat, other_feat,
```

File: torch_frame/typing.py [hunk 5]
```diff
@@ -66,4 +67,4 @@ def is_numerical_strategy(self):
 TextTokenizationMapping = Mapping[str, Tensor]
 TextTokenizationOutputs = Union[List[TextTokenizationMapping],
                                 TextTokenizationMapping]
-TensorData = Union[Tensor, MultiNestedTensor, Dict[str, MultiNestedTensor]]
+TensorData = Union[Tensor, _MultiTensor, Dict[str, MultiNestedTensor]]
```
Comment (zechengz) [inline torch_frame/typing.py:70]: Maybe union of MultiNestedTensor and MultiEmbeddingTensor is better?

**Review by zechengz — APPROVED** [review]

**Pull request merged**